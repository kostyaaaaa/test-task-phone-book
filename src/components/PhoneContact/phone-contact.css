.phone-contact {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;
}

.phone-contact-name {
  margin-left: 10px;
}

.phone-contact-phone {
  margin-left: 10px;
}

.phone-contact-delete {
  margin-left: 10px;
}
