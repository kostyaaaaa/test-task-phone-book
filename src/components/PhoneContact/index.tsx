import React, { useState } from "react";
import "./phone-contact.css";

type PhoneContactType = {
  contact: {
    name: string;
    phone: string;
    id: number;
  };
  deleteContact: (id: number) => void;
  updateContact: (contact: { phone: string; name: string; id: number }) => void;
};

const PhoneContact = ({
  contact: { name, phone, id },
  deleteContact,
  updateContact,
}: PhoneContactType) => {
  const [newName, setNewName] = useState<string>(name);
  const [newPhone, setNewPhone] = useState<string>(phone);

  const handleNewName = (event: React.ChangeEvent<HTMLInputElement>) => {
    setNewName(event.target.value);
  };

  const handleNewPhone = (event: React.ChangeEvent<HTMLInputElement>) => {
    setNewPhone(event.target.value);
  };

  const handleUpdateContact = () => {
    updateContact({ name: newName, phone: newPhone, id });
  };

  return (
    <div className="phone-contact">
      <input
        value={newName}
        onChange={handleNewName}
        className="phone-contact-name"
      />
      <input
        value={newPhone}
        onChange={handleNewPhone}
        className="phone-contact-phone"
      />
      {(name !== newName || newPhone !== phone) && (
        <button onClick={handleUpdateContact}>Save changes</button>
      )}
      <button
        className="phone-contact-delete"
        onClick={() => deleteContact(id)}
      >
        Delete contact
      </button>
    </div>
  );
};

export default PhoneContact;
