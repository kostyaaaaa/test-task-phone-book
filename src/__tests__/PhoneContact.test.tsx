import React from "react";
import PhoneContact from "../components/PhoneContact";
import { shallow } from "enzyme";

let wrapper: any;
const deleteContact = jest.fn();

describe("PhoneContact", () => {
  beforeEach(() => {
    wrapper = shallow(
      <PhoneContact
        updateContact={() => {}}
        contact={{ name: "John", phone: "123", id: 3 }}
        deleteContact={deleteContact}
      ></PhoneContact>
    );
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).not.toBeNull();
  });
  it("inputs should react on change event", () => {
    expect(wrapper.find(".phone-contact-name").props().value).toEqual("John");
    wrapper.find(".phone-contact-name").simulate("change", {
      target: { value: "new user name" },
    });
    expect(wrapper.find(".phone-contact-name").props().value).toEqual(
      "new user name"
    );

    expect(wrapper.find(".phone-contact-phone").props().value).toEqual("123");
    wrapper.find(".phone-contact-phone").simulate("change", {
      target: { value: "new user phone" },
    });
    expect(wrapper.find(".phone-contact-phone").props().value).toEqual(
      "new user phone"
    );
  });
  it("should call deleteContact function on click", () => {
    expect(deleteContact).toBeCalledTimes(0);
    wrapper.find(".phone-contact-delete").simulate("click");
    expect(deleteContact).toBeCalledTimes(1);
  });
});
