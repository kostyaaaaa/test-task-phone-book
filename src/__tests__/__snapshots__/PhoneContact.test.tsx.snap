// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneContact should render correctly 1`] = `
<div
  className="phone-contact"
>
  <input
    className="phone-contact-name"
    onChange={[Function]}
    value="John"
  />
  <input
    className="phone-contact-phone"
    onChange={[Function]}
    value="123"
  />
  <button
    className="phone-contact-delete"
    onClick={[Function]}
  >
    Delete contact
  </button>
</div>
`;
